### PR TITLE
feat: Propeller price evaluation.

### PR DESCRIPTION
**Pull request by tamaralipows**
Reason to use the provider instead of accessing redis directly:
- We will be able to tell if our logic in the provider is off (as opposed to just something in Redis).
- Less repeated code.

### CODE REVIEW

File: pyth_publisher/providers/propeller.py [hunk 1]
```diff
@@ -84,9 +90,18 @@ async def _update_loop(self) -> None:
 
     async def _update_prices(self) -> None:
         start = datetime.utcnow()
+
+        if self._quote_token not in self._supported_products:
+            log.warning(
+                "Quote token not supported, adding it to the supported products."
+            )
+            self._supported_products.add(self._quote_token.address)
```
Comment (dianacarvalho1) [inline pyth_publisher/providers/propeller.py:98]: why is this necessary?

File: pyth_publisher/providers/propeller.py [hunk 2]
```diff
@@ -65,6 +65,12 @@ def upd_products(self, product_symbols: List[PythSymbol]) -> List[PythSymbol]:
                     supported_products.append(product)
         return supported_products
 
+    def upd_products_internal(self, token_addresses: List[Address]):
+        """Update our provider with new products from our price evaluator."""
+        for address in token_addresses:
+            self._supported_products.add(address)
+        log.info(f"Added {len(self._supported_products)} tokens as supported products.")
```
Comment (dianacarvalho1) [inline pyth_publisher/providers/propeller.py:72]: so that when we add new tokens to the s3 file we don't need to reboot right?? veeeerryy niceeeeee ✨

**Review by dianacarvalho1 — APPROVED** [review]
Thank you @tamaralipows 🙏🏼🙏🏼 This looks reaally niceeee! happy we have this! Only have a small question

File: pyth_publisher/providers/propeller.py (hunk 1 above)
Comment (tamaralipows) [reply pyth_publisher/providers/propeller.py:98]: It's not. This is a part of my debugging because I thought it was missing here. T_T Stay tuned for the next commit.
